src/domain/products/product-form/sections/sales-channels.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useEffect } from "react"
+import { Product, SalesChannel } from "@medusajs/medusa"
+import { useAdminSalesChannels } from "medusa-react"
+import Badge from "../../../../components/fundamentals/badge"
+import Button from "../../../../components/fundamentals/button"
+import ChannelsIcon from "../../../../components/fundamentals/icons/channels-icon"
+import CrossIcon from "../../../../components/fundamentals/icons/cross-icon"
+import BodyCard from "../../../../components/organisms/body-card"
+
+type SalesChannelsProps = {
+  isEdit?: boolean
+  product: Product
+}
+
+type SalesChannelBadgeProps = {
+  channel: SalesChannel
+  onClick: () => void
+}
+
+const SalesChannels: React.FC<SalesChannelsProps> = ({
+  isEdit = false,
+  product,
+}) => {
+  const { count, isLoading } = useAdminSalesChannels()
+
+  const [remainder, setRemainder] = React.useState(0)
+
+  useEffect(() => {
+    if (product.sales_channels) {
+      setRemainder(Math.max(product.sales_channels.length - 3, 0))
+    }
+  }, [isLoading])
```

Comment:
suggestion: good practice is to always add all deps to prevent stale state issues
also this can be just a computed value?